documentation/api/wire_format/report_format_v6.markdown

```diff
@@ -0,0 +1,214 @@
+    }
+
+`"logs"` is either null or an array of log objects like so:
+
+    {
+      "file" : <file of resource declaration>,
+      "line" : <line of resource declaration>,
+      "level" : <log level>,
+      "message" : <log message>,
+      "source" : <log source>,
+      "tags" : [<resource tag>],
+      "time" : <log line timestamp>
+     }
+
+**Note on fields that allow `NULL` values**
+
+In the resource_event schema above, `containment_path`, `new_value`, `old_value`, `property`, `file`, `line`, `status`, and `message` may all be null.
+
+### Encoding
+
+The entire report is expected to be valid JSON, which implies UTF-8
+encoding.
+
+### Data type: `<string>`
+
+A JSON string. By virtue of the report being UTF-8, these must also
+be UTF-8.
+
+### Data type: `<integer>`
+
+A JSON int.
+
+### Data type: `<datetime>`
+
+A JSON string representing a date and time (with time zone), formatted based on
+the recommendations in ISO8601; so, e.g., for a UTC time, the String would be
+formatted as `YYYY-MM-DDThh:mm:ss.sssZ`.  For non-UTC, the `Z` may be replaced
+with `±hh:mm` to represent the specific timezone.
+
+### Data type: `<resource>`
+
+A JSON Object of the following form:
+
+
+    {
+      "timestamp": <datetime>,
+      "resource_type": <type>,
+      "resource_title": <title>,
+      "skipped" : <boolean>, 
+      "events" : [<event>, <event>, ...]
+      "file": <string>,
+      "line": <integer>,
+      "containment_path": [<string>, <string>, ...]
+    }
+
+where each `<event>` is of the form:
+
+    {
+     "property": <string>,
+     "timestamp": <datetime>,
+     "status": <string>,
+     "old_value": <string>,
+     "new_value": <string>,
+     "message": <string>,
+     "file": <string>,
+     "line: <integer>
+    }
+
+All keys are required.
+
+`"resource_type"` is the name of the Puppet resource type that the event occurred on.
+
+`"resource_title"` is the title of the Puppet resource that the event occurred on.
+
+`"timestamp"` is the date/time at which the resource's events occurred.
+
+`"status"` is a string representing the outcome of the event; this should be one
+of `success`, `failure`, or `noop`.
+
+`"skipped"` is a boolean representing whether or not the resource was skipped during the
+Puppet run; a skipped resource should have an empty list of `"events"`.
+
+`"containment_path"` is a collection of strings where each string is a Puppet type or class
+that represents the containment hierarchy of the resource within the catalog. This field may be `null`.
+
+`"events"` is a collection of objects where each is a Puppet event corresponding
+to the resource. This array may be empty.
+
+### Data type: `<event>`
+
+A JSON Object of the following form:
+
+
+    {
+     "property": <string>,
+     "timestamp": <datetime>,
+     "status": <string>,
+     "old_value": <string>,
+     "new_value": <string>,
+     "message": <string>,
+     "file": <string>,
+     "line: <integer>
+    }
```

Comment:
this same object is on line 151. We can probably delete it up there and leave it here.